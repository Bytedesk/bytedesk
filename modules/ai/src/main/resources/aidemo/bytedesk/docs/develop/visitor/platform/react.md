---
sidebar_label: React
sidebar_position: 2
---

# React

## 部分功能

- 微语官方技术支持
- 100%全部开源，支持自定义界面
- 支持人工客服
- 支持机器人
- 支持文字、图片、语音、表情
- 支持消息预知：对方正在输入
- 支持消息状态：送达、已读
- 支持消息撤回
- 对接第三方账号系统/多用户切换

## SDK源码及Demo下载

- [Gitee Demo](https://gitee.com/270580156/bytedesk-react)
- [Github Demo](https://github.com/Bytedesk/bytedesk-react)

## 安装

```bash
npm install bytedesk-react --save
# or
yarn add bytedesk-react
```

## 使用

```jsx
import { ChatFloat } from "bytedesk-react";
// 
<ChatFloat
    chatUrl="http://localhost:9006/chat?t=1&sid=default_wg_uid&"
    //buttonPosition: 'right', // botton position：left or right
    //buttonBackgroundColor: 'blue', // button background color
    //iframeWidth: 400,
    //iframeHeight: 600,
    //iframeMargins: { right: 20, bottom: 20, left: 20 }, // iframe margins
    //buttonMargins: { right: 20, bottom: 20, left: 20 }, // button margins
    //showButton: true, // show button or not
    //showIframe: true // show iframe or not
/>
// params:
chatUrl: 客服链接;
```

## 获取客服链接

- [登录管理后台](https://www.weiyuai.cn/admin/cs/wgroup)
