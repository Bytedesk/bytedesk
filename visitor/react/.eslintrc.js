module.exports = {
  // extends: [require.resolve('@umijs/lint/dist/config/eslint')],
  globals: {
    page: true,
    REACT_APP_ENV: true,
  },
  plugins: ["import"],
  rules: {
    // Note: you must disable the base rule as it can report incorrect errors
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": "off",
    "import/no-unused-modules": "off",
  },
};
