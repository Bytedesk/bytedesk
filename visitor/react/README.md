<!--
 * @Author: jackning 270580156@qq.com
 * @Date: 2023-11-14 16:59:09
 * @LastEditors: jackning 270580156@qq.com
 * @LastEditTime: 2024-04-05 12:17:32
 * @Description: bytedesk.com https://github.com/Bytedesk/bytedesk
 *   Please be aware of the BSL license restrictions before installing Bytedesk IM –
 *  selling, reselling, or hosting Bytedesk IM as a service is a breach of the terms and automatically terminates your rights under the license.
 *  仅支持企业内部员工自用，严禁私自用于销售、二次销售或者部署SaaS方式销售
 *  Business Source License 1.1: https://github.com/Bytedesk/bytedesk/blob/main/LICENSE
 *  contact: 270580156@qq.com
 * 联系：270580156@qq.com
 * Copyright (c) 2024 by bytedesk.com, All Rights Reserved.
-->

# 微语IM - 访客端

- [test](http://localhost:9006/?t=1&id=c293374081784c0e9dd1d57020048e37&)
- [ai](http://localhost:9006/ai?rid=20e4feffffb74f5eab46358e4d4d3bb3&)
- [pro](http://localhost:9006/pro)

```bash
# "@chatui/core": "^2.4.2",
export http_proxy=http://127.0.0.1:10818
export https_proxy=http://127.0.0.1:10818
# 打包
yarn build
# 代码格式化 https://www.prettier.cn/docs/install.html
# yarn prettier . --write
```

## other

- <https://chatui.io/sdk/getting-started>
