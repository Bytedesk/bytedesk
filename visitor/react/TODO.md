<!--
 * @Author: jackning 270580156@qq.com
 * @Date: 2024-04-03 17:20:03
 * @LastEditors: jackning 270580156@qq.com
 * @LastEditTime: 2024-04-05 14:48:15
 * @Description: bytedesk.com https://github.com/Bytedesk/bytedesk
 *   Please be aware of the BSL license restrictions before installing Bytedesk IM –
 *  selling, reselling, or hosting Bytedesk IM as a service is a breach of the terms and automatically terminates your rights under the license.
 *  仅支持企业内部员工自用，严禁私自用于销售、二次销售或者部署SaaS方式销售
 *  Business Source License 1.1: https://github.com/Bytedesk/bytedesk/blob/main/LICENSE
 *  contact: 270580156@qq.com
 * 联系：270580156@qq.com
 * Copyright (c) 2024 by bytedesk.com, All Rights Reserved.
-->

# TODO:

- 国际化
<!-- - 支持深色主题 -->
- 支持访客模式
- 存储cookie信息
<!-- - 支持登录模式 -->
- 支持客服、工单、意见反馈、问卷、知识库
- 如果检测到实在app内部调用h5，则可调用手机相册和录音
- 获取浏览器notification权限，推送通知
- 播放新消息提示音
- 客服插件可对web页面进行截图或选择部分区域，并发送给客服
