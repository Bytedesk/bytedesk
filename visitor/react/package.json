{
  "name": "visitor",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "start": "react-scripts start",
    "dev": "react-scripts start",
    "serve": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "dependencies": {
    "@ant-design/icons": "^5.3.6",
    "@stomp/stompjs": "^7.0.0",
    "@testing-library/jest-dom": "^5.17.0",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^13.5.0",
    "@types/jest": "^27.5.2",
    "@types/node": "^16.18.61",
    "@types/react": "^18.2.15",
    "@types/react-dom": "^18.2.15",
    "@use-gesture/react": "^10.3.0",
    "antd": "^5.16.0",
    "axios": "^1.6.2",
    "clsx": "^2.0.0",
    "dompurify": "^3.0.6",
    "event-source-polyfill": "^1.0.31",
    "i18next": "^23.10.1",
    "i18next-browser-languagedetector": "^7.2.1",
    "intersection-observer": "^0.12.2",
    "mitt": "^3.0.1",
    "react": "^18.2.0",
    "react-cookie": "^7.1.4",
    "react-dom": "^18.2.0",
    "react-dropzone": "^14.2.3",
    "react-helmet": "^6.1.0",
    "react-i18next": "^14.1.0",
    "react-router-dom": "^6.18.0",
    "react-scripts": "5.0.1",
    "sockjs-client": "^1.6.1",
    "typescript": "^4.9.5",
    "ua-parser-js": "^1.0.37",
    "web-vitals": "^2.1.4",
    "zustand": "^4.4.4"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@babel/plugin-proposal-private-property-in-object": "^7.21.11",
    "@types/ua-parser-js": "^0.7.39",
    "prettier": "3.2.5"
  }
}
