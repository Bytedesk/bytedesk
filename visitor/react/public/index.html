<!--
 * @Author: jackning 270580156@qq.com
 * @Date: 2023-11-14 16:59:09
 * @LastEditors: jackning 270580156@qq.com
 * @LastEditTime: 2024-03-23 12:49:49
 * @Description: bytedesk.com https://github.com/Bytedesk/bytedesk
 *   Please be aware of the BSL license restrictions before installing Bytedesk IM – 
 *  selling, reselling, or hosting Bytedesk IM as a service is a breach of the terms and automatically terminates your rights under the license. 
 *  仅支持企业内部员工自用，严禁私自用于销售、二次销售或者部署SaaS方式销售 
 *  Business Source License 1.1: https://github.com/Bytedesk/bytedesk/blob/main/LICENSE 
 *  contact: 270580156@qq.com 
 * 联系：270580156@qq.com
 * Copyright (c) 2024 by bytedesk.com, All Rights Reserved. 
-->
<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="utf-8" />
  <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="theme-color" content="#000000" />
  <meta name="description" content="Bytedesk AI - Customer Service with AI" />
  <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
  <!--
      manifest.json provides metadata used when your web app is installed on a
      user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
    -->
  <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
  <!--
      Notice the use of %PUBLIC_URL% in the tags above.
      It will be replaced with the URL of the `public` folder during the build.
      Only files inside the `public` folder can be referenced from the HTML.

      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
      work correctly both with client-side routing and a non-root public URL.
      Learn how to configure a non-root public URL by running `npm run build`.
    -->
  <title>微语</title>
</head>

<body>
  <noscript>You need to enable JavaScript to run this app.</noscript>
  <div id="root"></div>
  <!--
      This HTML file is a template.
      If you open it directly in the browser, you will see an empty page.

      You can add webfonts, meta tags, or analytics to this file.
      The build step will place the bundled scripts into the <body> tag.

      To begin the development, run `npm start` or `yarn start`.
      To create a production bundle, use `npm run build` or `yarn build`.
    -->

  <!-- https://chatui.io/sdk/getting-started -->
  <!-- 默认输入框右侧图标不显示，需要添加下面js -->
  <script src="./assets/js/0.3.5/sdk.js"></script>
  <script src="./assets/js/5.15.3/isv-parser.js"></script>
  <script src="./assets/js/2.0.2/index.js" async></script>
</body>

</html>