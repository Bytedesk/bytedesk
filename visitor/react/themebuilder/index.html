<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <title>ChatUI Theme Builder</title>
    <!-- <link rel="stylesheet" href="//g.alicdn.com/chatui/theme-builder/0.3.0/index.css"> -->
    <link rel="stylesheet" href="./index.css" />
  </head>

  <body>
    <script>
      <!-- with (document) with (body) with (insertBefore(createElement("script"), firstChild)) setAttribute("exparams", "category=&userid=&aplus&yunid=&b5ee331eeee3&asid=AQAAAACLHK9lsSIYeQAAAABClh5DA1BZeg==", id = "tb-beacon-aplus", src = (location > "https" ? "//g" : "//g") + ".alicdn.com/alilog/mlog/aplus_v2.js") -->
      with (document)
        with (body)
          with (insertBefore(createElement("script"), firstChild))
            setAttribute(
              "exparams",
              "category=&userid=&aplus&yunid=&b5ee331eeee3&asid=AQAAAACLHK9lsSIYeQAAAABClh5DA1BZeg==",
              (id = "tb-beacon-aplus"),
              (src = "./aplus_v2.js"),
            );
    </script>
    <div id="root"></div>
    <script
      src="//g.alicdn.com/code/lib/??react/17.0.2/umd/react.production.min.js,react-dom/17.0.2/umd/react-dom.production.min.js"
      crossorigin
    ></script>
    <!-- <script src="//g.alicdn.com/chatui/theme-builder/0.3.0/index.js" crossorigin></script> -->
    <script src="./index.js" crossorigin></script>
  </body>
</html>
