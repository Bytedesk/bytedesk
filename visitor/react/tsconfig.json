{
  "compilerOptions": {
    "target": "esnext",
    "module": "esnext",
    "moduleResolution": "node",
    "importHelpers": true,
    "jsx": "preserve",
    "esModuleInterop": true,
    "sourceMap": true,
    "skipLibCheck": true,
    "experimentalDecorators": true,
    "strict": false,
    "resolveJsonModule": true,
    "allowSyntheticDefaultImports": true,
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "isolatedModules": false,
    "noEmit": true,
    "baseUrl": "./",
    "paths": {
      "@/*": ["src/*"]
    }
  },
  "include": ["src"]
}
