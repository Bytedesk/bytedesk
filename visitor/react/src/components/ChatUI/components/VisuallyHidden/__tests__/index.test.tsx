import React from "react";
import { render, cleanup } from "@testing-library/react";

import { VisuallyHidden } from "..";

afterEach(cleanup);

describe("<VisuallyHidden />", () => {
  it("should render correctly", () => {
    const { container } = render(<VisuallyHidden />);
    expect(container).toMatchSnapshot();
  });
});
