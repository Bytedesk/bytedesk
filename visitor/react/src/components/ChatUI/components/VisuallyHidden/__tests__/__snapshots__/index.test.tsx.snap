// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VisuallyHidden /> should render correctly 1`] = `
<div>
  <span
    style="position: absolute; height: 1px; width: 1px; overflow: hidden; margin: -1px; white-space: nowrap;"
  />
</div>
`;
